priam/src/main/java/com/netflix/priam/identity/InstanceIdentity.java

```diff
@@ -109,154 +117,162 @@ public InstanceInfo getInstanceInfo() {
+                        if (!StringUtils.isEmpty(replaceIp)
+                                && !replaceIp.equals(instance.getHostIP())) {
+                            setReplacedIp(replaceIp);
                         }
-                    }.call();
-        }
+                    }
+                }
 
-        // Grab a new token
-        if (null == myInstance) {
-
-            if (this.config.isCreateNewTokenEnable()) {
-
-                myInstance =
-                        new RetryableCallable<PriamInstance>() {
-
-                            @Override
-                            public PriamInstance retriableCall() throws Exception {
-                                super.set(100, 100);
-                                newTokenRetriever.setLocMap(locMap);
-                                return newTokenRetriever.get();
-                            }
-
-                            @Override
-                            public void forEachExecution() {
-                                populateRacMap();
-                                newTokenRetriever.setLocMap(locMap);
-                            }
-                        }.call();
-
-            } else {
-                throw new IllegalStateException(
-                        "Node attempted to erroneously create a new token when we should be grabbing an existing token.");
+                if (instance != null) {
+                    logger.info(
+                            "{} found that this node is {}."
+                                    + " application: {},"
+                                    + " id: {},"
+                                    + " instance: {},"
+                                    + " region: {},"
+                                    + " host ip: {},"
+                                    + " host name: {},"
+                                    + " token: {}",
+                            instance.isOutOfService() ? "[Dead]" : "[Alive]",
+                            instance.isOutOfService() ? "dead" : "alive",
+                            instance.getApp(),
+                            instance.getId(),
+                            instance.getInstanceId(),
+                            instance.getDC(),
+                            instance.getHostIP(),
+                            instance.getHostName(),
+                            instance.getToken());
+                }
+
+                return instance;
+            }
+        }.call();
+    }
+
+    private PriamInstance mayGrabDeadToken() throws Exception {
+        return new RetryableCallable<PriamInstance>() {
+            @Override
+            public PriamInstance retriableCall() throws Exception {
+                PriamInstance result = deadTokenRetriever.get();
+                if (result != null) {
+                    isReplace = true; // indicate that we are acquiring a dead instance's token.
+
+                    if (deadTokenRetriever.getReplaceIp()
```

Comment:
This is taken care of in `grabAlreadyAssignedToken()` method. It returns the instance and C* is started in bootstrap mode.

Reply:
I do not agree with this statement. From infer token ownership in DeadToken we can return null. 
`logger.warn(
                "Return null: Unable to find enough instances where gossip match. Required: {}",
                noOfInstancesGossipShouldMatch);
        return null;`

In that case, DeadTokenRetriever will return null and thus no IP will be set in replacedIp. We overcome this particular thing in NFDeadTokenRetriever and my goal was to bring that goodness here as well with enough documentation.

Reply:
I modified TokenRetrievalUtils to throw GossipParseException when it is not able to fetch the gossip info or not able to parse the token info. With the exception we could differentiate gossip mismatch with gossip error. In case of gossip errror, the deadTokenRetriever falls back to the IP from token database.